accepted/0000-switching-registries.md

```diff
@@ -0,0 +1,63 @@
+
+# Motivation
+
+The default registry used by Yarn is `registry.yarnpkg.com`, but the use of alternate registries is possible through configuration or command-line flags. Using an alternate registry can be useful for a number of reasons; it can improve performance, facilitate serving private packages, and improve the reliability of build and test servers. However, many of these use cases are environment specific. These benefits cannot be realized without allowing the use of different registries in different environments.
+
+* Using a registry mirror that is closer to you can dramatically improve performance. Using a single registry mirror might work well for teams that work in the same office, but teams that span great distances will need to use different registry mirrors to see the same benefit.
+* Build environments can benefit from using a dedicated registry. In addition to the performance benefits, it can also protect against outages that might affect the public registries.
+
+Currently, Yarn will only use the configured registry for newly installed packages. For packages that are listed in the lockfile already, Yarn will use the registry saved in the lockfile instead, ignoring the registry configuration of the current user. This effectively prevents switching registries between environments.
+
+# Detailed design
+
+Yarn should adopt the behavior `npm` introduced with `npm@5`, which is to always use the registry configured by the current user. This change was described in [this blog post](http://blog.npmjs.org/post/161081169345/v500):
+>If you generated your package lock against registry A, and you switch to registry B, npm will now try to install the packages from registry B, instead of A. If you want to use different registries for different packages, use scope-specific registries (npm config set @myscope:registry=https://myownregist.ry/packages/). Different registries for different unscoped packages are not supported anymore.
+
+Yarn already supports switching to a different registry and scoped registries. The change would be to use them in all cases rather than just for new packages.
+
+### What about the `resolved` field in the lockfile?
+
+The `resolved` field will not be used to determine which registry is used. Changes to to the `resolved` field, such as removing the registry, are outside the scope of this RFC.
+
+### How do scoped registries work?
+
+Yarn supports configuring a registry for a specific scope (e.g. `yarn config set '@foo:registry' 'https://registry.foo.com'`). If a scoped registry has been configured, then this registry shall be used for **all** packages under that scope. Using different registries within a single scope is not supported.
+
+### Can non-scoped packages use alternate registries?
+
+No, non-scoped packages would all use the same registry. This restriction simplifies the configuration, and keeps us in line with `npm`.
```

Comment:
Well, they can use alternate registries, but on a "all or nothing basis". The technical answer is "yes" but your explanation is good 😛

Reply:
Ah.. right 😛  I shall edit this to make that more clear.